config.go

```diff
@@ -0,0 +1,48 @@
+package wireless
+
+import (
+	"encoding/hex"
+
+	"github.com/pkg/errors"
+)
+
+// https://github.com/digsrc/wpa_supplicant/blob/515eb37dd1df3f4a05fc2a31c265db6358301988/src/utils/common.c#L631-L685
+func configParseString(s string) (string, error) {
+	l := len(s)
+	if l == 0 {
+		return "", nil
+	}
+	if s[0] == '"' {
+		if s[l-1] != '"' {
+			return "", errors.New("Quoted string did not end with a quote character")
+		}
+		return s[1 : l-1], nil
+	} else if l > 1 && s[0] == 'P' && s[1] == '"' {
+		if s[l-1] != '"' {
+			return "", errors.New("printf_format string did not end with a quote character")
+		}
+		return DecodeSsid(s[2 : l-1])
+	} else {
+		// If the config value does not start with a quote character, it should be hex-encoded
+		v, err := hex.DecodeString(s)
+		return string(v), err
+	}
+}
+
+func isHex(s string) bool {
+	for _, c := range s {
+		if c < 32 || c >= 127 {
+			return true
+		}
```

Comment:
it's not clear what this is doing. if one char is in the range 32,127 then it is a hex?

```
In [7]: ''.join([chr(i) for i in range(32, 127)])
Out[7]: ' !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~'
```

Reply:
Updated here: https://github.com/Plantiga/go-wireless/pull/2/commits/ae774a838736be5f46dd702704e1696e87a03dad 

This is the function name used in wpa_supplicant - basically if the character is outside this range, the config value is hex-encoded instead of a string.